components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/processors/SPInitLogoutRequestProcessor.java

```diff
@@ -174,14 +174,10 @@ public SAMLSSOReqValidationResponseDTO process(LogoutRequest logoutRequest, Stri
     private void setX509Certificate(String issuer, SAMLSSOServiceProviderDO logoutReqIssuer) {
 
         try {
-            String tenant = logoutReqIssuer.getTenantDomain();
-            ServiceProvider serviceProvider = SAMLSSOUtil.getApplicationMgtService().
-                    getServiceProviderByClientId(issuer, SAMLSSOConstants.INBOUND_AUTH_TYPE_SAML, tenant);
-            String cert = serviceProvider.getCertificateContent();
-            X509Certificate certificate = (X509Certificate) IdentityUtil.convertPEMEncodedContentToCertificate(cert);
-            logoutReqIssuer.setX509Certificate(certificate);
-
-        } catch (IdentityApplicationManagementException | CertificateException e) {
+            SAMLSSOServiceProviderDO serviceProviderConfigs = getServiceProviderConfig(issuer);
```

Comment:
If the service provider certificate defined at the service provider level. Will this handle that?

Reply:
this follows the same approach as in SAML SSO. This considers certificate which was configured when creating the SAML service provider(under cert alias)